### PR TITLE
Release/snowplow fractribution/0.3.2

### PR DESCRIPTION
**Pull request by agnes-kiss**
## Description & motivation
This version gives the option to the user to filter on the timestamp field their events table is partitioned on for query optimization purposes. Unless changed, the `conversions_by_customer_id` model will filter based on `derived_tstamp`.

## Checklist
- [x] I have verified that these changes work locally
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (Link here if required)

## Release Only Checklist
- [x] I have updated the version number in all relevant places
- [x] I have updated the CHANGELOG.md

### CODE REVIEW

**Review by rlh1994 — APPROVED** [review]

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,14 @@
+snowplow-fractribution 0.3.2 (2023-07-25)
+---------------------------------------
+## Summary
+This version gives the option to the user to filter on the timestamp field their events table is partitioned on for query optimization purposes. Unless changed, the `conversions_by_customer_id` model will filter based on `derived_tstamp`.
```
Comment (zloff) [inline CHANGELOG]: I found this first sentence a bit hard to follow - maybe something like:
"This version allows the user to filter their events table using the timestamp field that the table is partitioned on, optimizing query performance."

File: models/snowplow_fractribution_conversions_by_customer_id.sql [hunk 2]
```diff
@@ -24,3 +24,8 @@ from {{ var('snowplow__conversions_source' )}} as events
 where {{ conversion_clause() }}
   and date(derived_tstamp) >= '{{ get_lookback_date_limits("min") }}'
   and date(derived_tstamp) <= '{{ get_lookback_date_limits("max") }}'
+
+  {% if var('snowplow__conversions_source_filter') != '' %}
+    and date({{ var('snowplow__conversions_source_filter') }}) >= {{ dateadd('day',-var('snowplow__conversions_source_filter_buffer_days'), "'"~get_lookback_date_limits('min')~"'") }}
+    and date({{ var('snowplow__conversions_source_filter') }}) <= {{ dateadd('day', var('snowplow__conversions_source_filter_buffer_days'),"'"~get_lookback_date_limits('max')~"'") }}
+  {% endif %}
```
Comment (zloff) [inline models/snowplow_fractribution_conversions_by_customer_id.sql:31]: I wonder if we should change this to an `if/else`, to something like:
```
where {{ conversion_clause() }}

  {% if var('snowplow__conversions_source_filter') == '' %}
  and date(derived_tstamp) >= '{{ get_lookback_date_limits("min") }}'
  and date(derived_tstamp) <= '{{ get_lookback_date_limits("max") }}'

{% else %}
    and date({{ var('snowplow__conversions_source_filter') }}) >= {{ dateadd('day',-var('snowplow__conversions_source_filter_buffer_days'), "'"~get_lookback_date_limits('min')~"'") }}
    and date({{ var('snowplow__conversions_source_filter') }}) <= {{ dateadd('day', var('snowplow__conversions_source_filter_buffer_days'),"'"~get_lookback_date_limits('max')~"'") }}
  
{% endif %}
```
Otherwise I think we may end up excluding events that the user didn't want excluded, e.g.:
| event_name | derived_tstamp | collector_tstamp |
| page_view  | 2023-01-01 23:59:00 | 2023-01-02 00:01:00 |

if the `min_date` was `2023-01-02` and they add `collector_tstamp` as their `snowplow__conversions_source_filter`, this event will be excluded in the current code, but not in the if/else code

File: dbt_project.yml [hunk 3]
```diff
@@ -40,7 +40,8 @@ vars:
     # Overwrite these source table vars in your own dbt_project.yml if not using these defaults:
     snowplow__page_views_source: "{{ source('derived', 'snowplow_web_page_views') }}"
     snowplow__conversions_source: "{{ source('atomic', 'events') }}"
-
+    snowplow__conversions_source_filter: '' # a timestamp field the conversion source field is partitioned on (ideally) for optimized filtering, leave blank to disable
```
Comment (zloff) [inline dbt_project.yml]: Maybe instead of `, leave blank to disable`, mention that if it is blank, the default of `derived_tstamp` will be used?

File: models/snowplow_fractribution_conversions_by_customer_id.sql (hunk 2 above)
Comment (zloff) [reply models/snowplow_fractribution_conversions_by_customer_id.sql:31]: I just noticed `snowplow__conversions_source_filter_buffer_days` - is that in order to get around this problem? Maybe i'm missing something, but I think removing the extra variable and just adding an if/else might be less complicated? also we can't guarantee that the buffer will prevent any events being dropped that shouldn't

**Review by zloff — COMMENTED** [review]
Allowing optimisation is a great idea! Just a couple of comments

File: models/snowplow_fractribution_conversions_by_customer_id.sql (hunk 2 above)
Comment (rlh1994) [reply models/snowplow_fractribution_conversions_by_customer_id.sql:31]: I think the issue with using just one is that we rely on that field elsewhere in the code don't we? So we expect a certain date range based on derived? Or am I misremembering

Comment (zloff) [reply models/snowplow_fractribution_conversions_by_customer_id.sql:31]: hmmm yeah good point, it is also a filter in `sessions_by_customer_id` which queries the page_views table.

I'm not super familiar with the snowplow web package - is this table always partitioned on `derived_tstamp`? And nevertheless, is it something we should consider if the filter clause on the conversions table is different to the filter for the page_views table?

Comment (rlh1994) [reply models/snowplow_fractribution_conversions_by_customer_id.sql:31]: Yeah page views is always derived unless they do some custom stuff with it. That's why we went with the buffer approach to try and ensure the same records but just a more optimised query

File: CHANGELOG (hunk 1 above)
Comment (agnes-kiss) [reply CHANGELOG]: Yeah, sounds much better, thanks!
